purchases/src/main/kotlin/com/revenuecat/purchases/Purchases.kt

```diff
@@ -1222,12 +1222,12 @@ class Purchases internal constructor(
         }
     }
 
-    private fun getSkuDetails(
+    private fun getProductGroups(
         productIds: Set<String>,
-        onCompleted: (HashMap<String, StoreProduct>) -> Unit,
+        onCompleted: (Map<String, StoreProduct>) -> Unit,
         onError: (PurchasesError) -> Unit
     ) {
-        billing.querySkuDetailsAsync(
+        billing.queryProductDetailsAsync(
```

Comment:
the problem is that we now might have multiple products for a subscription ID (a single sub id may have multiple base plans), so when we group by `subProduct.sku` we lose all but one of those

Reply:
I am not sure I understand this comment. Can you give me an example? Thanks!

Reply:
sorry that wasn't very clear. i mean that the call to queryProductDetails is fetching a _subscription_, which can have multiple base plans. we are loosely mapping a base plan to our StoreProduct, so this `onCompleted` is the subscriptionId -> our store products, which i think needs to be a list so that it can contain all of the products.

for instance, we queryProductDetails for `onemonth_freetrial`, which has two base plans, `bc5-annual-renews-3-months` and `p1m`. when we call billing.queryProductDetailsAsync, we return a listOf(StoreProduct(bc5-annual-renews-3-months, p1m), both with the subscriptionId `onemonth_freetrial`

but then this code in Purchases `subscriptionProducts.associateBy { subProduct -> subProduct.sku }` creates Map<String, StoreProduct>, so we are losing one of them. 

(associateBy's docs say "If any two elements would have the same key returned by keySelector the last one gets added to the map"). 

does that make sense?

Reply:
It makes sense, thanks for explaining. Let's fix it with your work on https://github.com/RevenueCat/purchases-android/pull/666/